### ISSUE TITLE
Change Request: Remove required arguments from updateQuestion Mutation

### ISSUE THREAD

**Issue by ghost** [chingu-voyages/v8-chimeras-team-02#37]
The mutation requires: _id, title, question, createAt, user_id (see last remark for issue with user_id), tags, and answers_ids. I think that just the _id and user_id would be enough. Do you agree? (I couldn't test this one) - Ed

**Comment by ghost** [comment]
Looking at the code it's not difficult to change, and for things like createdAt your just passing it from the record itself.

We can do a check for only getting an _id and throwing a 'nothing to change' error, or return 'successfully changed nothing'

**Comment by etorres07** [comment]
Let me see if I understood correctly.

Basically, what you are saying is that:
1. You agree with my proposal on reducing the required fields for the `updateQuestion` mutation to only require the `_id` (which is the question's id)
2. You suggest adding a "catch" in the case that an erroneous or non-existent `_id` is passed into the mutation

Please correct me where I'm wrong or if I missed something.

**Comment by ghost** [comment]
You're wrong on number 2. I was thinking about a custom message for a call with only an _id, but no other arguments that tell the function _what_ to mutate. 

The "catch" you're suggesting isn't a bad idea though, we can use the catch from deleteQuestion.

**Comment by etorres07** [comment]
I'm having trouble following. ELI5, please.

What do you mean by "a custom message for a call"? What you're saying is that if we only provide an `_id` there might be some trouble because the server might not understand what we're asking for?

**Comment by ghost** [comment]
Alright, ELI5:

The function called `updateQuestion()` is used to talk to our database. At the moment that function wants all the parts of a question before it goes and talk to our database. This isn't necessary, it technically just needs an `_id` to find the question, the `user_id` and token to check if this user can update the question, and something to change. 

There is no easy way for the `updateQuestion()` to ask for `just the _id **and** at least one of these other things: [title, question, tags, etc]`. We have to choose which of them are `required` and which are not.

You're suggesting that we only require the _id and leave the rest up to the caller, which creates a new situation: calling `updateQuestion()` with just the _id. What should the function do? It can:
* take the _id, find the question and update all it has to update (nothing) and come back successfully.
or 
* see that it only has an `_id` and nothing to update and throw an error "I need something to update" so we can inform the user that something needs to change.

Which of those two we choose depends on how we're implementing updating questions in the front-end.

**Comment by etorres07** [comment]
Thanks for the explanation and sorry for the delay.

You say:

_What should the function do? It can:_
- *take the _id, find the question and update all it has to update (nothing) and come back successfully*

Why "nothing"? The idea behind `updateQuestion()` is to update the requested question with a new or modified one.

**Comment by ghost** [comment]
Yes, but what should happen if the function is called without a new or modified question. What if we call it with _just_  the _id and nothing else?

**Comment by etorres07** [comment]
I will close and reopen if necessary.
